### ISSUE TITLE
Further adaptations to `DownstreamExpirationTime` vs. `TransactionMaxLifetime`

### ISSUE THREAD

**Issue by DDvO** [siemens/LightweightCmpRa#9]
This is a follow-up on #7 und #8.

After our clarifying recent discussion, please further tweak the implementation and doc as follows.

* Rename `TransactionMaxLifetime` (and the leftover parameter/variable name `downstreamExpirationTime`) to  `DownstreamTimeout`.
*  The RA component sets up the internal time limit of a transaction in the moment when returning a response to downstream if it remembers the transaction because it expects further (polling or other) requests from downstream. 
If the response contains a `retryAfter` value, this value is added to the configured timeout value (such that the time limit is extended by the expected waiting time of the client).
* Adapt the documentation in the configuration README file accordingly, for instance like this.
```
The **`DownstreamTimeout` object**
optionally specifies the maximum allowed reaction time of the downstream entity
apart from any applicable `retryAfter` period.
That is, the RA will keep a transaction open while awaiting a further request
form the client side until it receives the expected request or the configured
number of seconds, plus any `retryAfter` time given in its last response, has elapsed.
In the latter case the RA cleans up and forgets the transaction.<br>
By default, or if the value 0 is given, no restriction is placed,
such that a transaction may idle indefinitely. It is recommended to specify
a nonzero timeout value in order to prevent blocking RA resources indefinitely,
for instance when an expected subsequent request message by the client is lost
or the client terminates during a transaction without the RA knowing.
```
* Adapt the documenting comment(s) in the respective CMP RA Component API Java file.

**Comment by Akretsch** [comment]
How ist the expected behaviour, if no retryAfter is in place?

**Comment by Akretsch** [comment]
fixed in a622075a25195c67ea6e3b902bb60fbd83d7d6b1
